modes/parkour/moderation.lua

```diff
@@ -0,0 +1,447 @@
+			ban_request[player] = nil
+			return
+		end
+
+		if fetch[3] == "resetbancount" then
+			data.bancount = 0
+			system.savePlayerData(player, data)
+			tfm.exec.chatMessage("<v>[#] <r>"..player.. "'s bans has been reset", fetch[1]) 
+			ban_request[player] = nil
+			return
+		end
+
+		local isban = fetch[3]
+		local requester = fetch[1]
+		if isban then
+			banPlayer(player, data, requester, true)
+		else
+			banPlayer(player, data, requester, false)
+		end
+		ban_request[player] = nil
+	end
+end
+
+function showChatCommands(p, command, args)
+	local commandtext = table.concat(args, " ")
+	for playername, player in pairs(tfm.get.room.playerList) do
+		if ranks.admin[playername] or ranks.mod[playername] then
+			tfm.exec.chatMessage("<BL>Ξ ["..p.."]<N2> !"..command.." "..commandtext, playername)
+		end
+	end
+end
+
+onEvent("ParsedChatCommand", function(player, cmd, quantity, args)
+	local max_args = quantity
+	if cmd == "ban" or cmd == "unban" then -- for mods / !ban Username and !unban Username 
+		if not ranks.admin[player] and not perms[player].ban then return end
+
+		if quantity < 1 then 
+			return translatedChatMessage("invalid_syntax", player)
+		end
+		
+		requestplayer = capitalize(args[1])
+		if not string.find(requestplayer, "#", 1, true) then
+			requestplayer = requestplayer .. "#0000"
+		end
+
+		showChatCommands(player, cmd, args)
+
+		if players_file[args[1]] then 
+			banPlayer(args[1], players_file[args[1]],player, cmd == "ban")
+		else
+			ban_request[args[1]] = {player, os.time() + 1000, cmd == "ban"}
+			system.loadPlayerData(args[1])
+		end
+
+	elseif cmd == "pban" or cmd == "punban" then -- only admins / !pban [username or id] [minutes or 1 for perm] - !punban [username or id]
+		if not ranks.admin[player] then return end
+
+		if quantity < 1 then 
+			return translatedChatMessage("invalid_syntax", player)
+		end
+
+		local minutes = args[2]
+		if not tonumber(args[1]) then 
+			local p = args[1] and tfm.get.room.playerList[args[1]]
+			playerid = p and p.id or nil
+
+			if playerid == nil then 
+				return tfm.exec.chatMessage("<v>[#] <r>You must be in the same room as the player to ban them using their username. \n<v>[>] <J>/c Recordsbot#8598 id "..args[1], player)
+			end
+		else
+			playerid = args[1]
+		end
+
+		if cmd == "punban" then 
+			schedule("ban_change", playerid, 0)
+			tfm.exec.chatMessage("<v>[#] <r>"..playerid.. " has unbanned.", player)
+		else
+			if minutes == 1 then
+				schedule("ban_change", playerid, 1)
+				return tfm.exec.chatMessage("<v>[#] <r>"..playerid.. " has permbanned.", player)
+			else
+				if not tonumber(minutes) then 
+					return tfm.exec.chatMessage(minutes .. " doesnt look like a number ", player)
+				end
+				minutes = os.time()+tonumber(minutes)*60*1000
+				schedule("ban_change", playerid, minutes)
+				tfm.exec.chatMessage("<v>[#] <r>" ..playerid.. " has banned for " ..(tonumber(args[2])).. " minutes", player)
```

Comment:
There seems to be too much repeated code. Have you considered splitting it into functions?